_docs/ux-design.md

```diff
@@ -0,0 +1,344 @@
+   &ndash; but has the disadvantage of making hermetic or reproductible builds
+   more difficult (the `go.mod` and `go.sum` files ought to be considered as
+   build artifacts, as they will be modified at the start of the build and are
+   needed to correctly reproduce a build in the future).
+
+### Building Applications
+
+Once the configuration has been created, either by the command-line assistant,
+or directly by the user (possibly through automated processes), the tool can be
+used directly to build, run, and test go applications directly:
+
+1. If the tool is installed as a Go `tool` dependency (`go1.24` and newer):
+
+   ```console
+   $ go tool gotel build -o bin/app .
+   $ go tool gotel test -shuffle=on ./...
+   ```
+
+2. Installing `gotel` in `$GOBIN`
+
+   ```console
+   $ go install github.com/open-telemetry/opentelemetry-go-compile-instrumentation/cmd/gotel
+   $ gotel build -o bin/app
+   $ gotel test -shuffle=on ./...
+   ```
+
+3. Running `gotel` with `go run`:
+
+   ```console
+   $ go run github.com/open-telemetry/opentelemetry-go-compile-instrumentation/cmd/gotel build -o bin/app
+   $ go run github.com/open-telemetry/opentelemetry-go-compile-instrumentation/cmd/gotel test -shuffle=on ./...
+   ```
+
+
+### Ongoing Maintenance
+
+Since the tool and configuration are registered in the `go.mod` file, users are
+able to keep these dependencies up-to-date using the standard tools and
+processes they use for any other dependency.
+
+They are also able to modify what confguration gets applied by changing the
+configuration according to their set up style, either:
+
+- directly in the `go.mod` file by adding or removing `tool` dependencies,
+- in the `gotel.instrumentation.go` file by adding or removing `import`
+  declarations.
+
+### Custom Configuration
+
+Users may wish to add their own, application-specific automatic instrumentation
+configuration. This is achieved by adding a `.gotel.yml` file in the same
+directory as the application's `go.mod` file. Such instructions will be used
+only when building packages in this module's context (and not when themodule's
+packages are dependencies of an automatically-instrumented application).
+
+The contents of such configuration files uses the same schema as the
+[instrumentation packages](#instrumentation-packages) definitions file.
+
+### Uninstalling
+
+Removing auto-instrumentation configuration is as simple as removing the related
+tool dependencies from `go.mod` and removing the `gotel.instrumentation.go`
+file.
+
+## Instrumentation Packages
+
+A majority of users of the OpenTelemetry compile-time instrumentation tool will
+rely on instrumentation packages to instrument their application. These are
+standard Go packages that are part of a Go module and contain either (or both):
+
+- a `gotel.instrumentation.yml` file that declares all instrumentation
+  configuration that is vended by this package;
+- a `gotel.instrumentation.go` file that imports at least one valid
```

Comment:
Could you elaborate more about how this file may looks like (Since you have introduced `gotel.instrumentation.yml`)?